gateway/api.go

```diff
@@ -1111,6 +1169,82 @@ func (gw *Gateway) apiHandler(w http.ResponseWriter, r *http.Request) {
 	doJSONWrite(w, code, obj)
 }
 
+func (gw *Gateway) apiOASGetHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		apiID       = mux.Vars(r)["apiID"]
```

Comment:
Code is duplicating a lot, create a function that takes a *Request and returns string with api id. This way we can use the compiler-time checking to avoid possible typos in map keys (what if it's sometimes "apiId"?)

Reply:
the param apiID is a constant name and checked by mux, I'm not sure we would need to validate anything on the name

Reply:
That doesn't really satisfy the concern. Code is duplicated with copy paste. The same logic is re-implemented in several places. Modifying it also means modifying it in several places. Using a string literal is subject to human errors, that's why a /single/ function to retrieve the apiID parameter from *Request is safer.

Reply:
this code is a single line(a constant unchanging single line), putting a single line in a function would not really solve duplicity but rather introduce redundancy

Reply:
I count 4 repeatitions in the PR, not one.